src/node/internal/internal_timers_promises.ts

```diff
@@ -62,16 +64,23 @@ export async function setTimeout<T = void>(
   }
 
   const { promise, resolve, reject } = Promise.withResolvers<T>();
+  let onCancel: OnCancelCallback;
 
   const timer = timers.setTimeout(() => {
     resolve(value as T);
+    if (onCancel) {
+      signal?.removeEventListener('abort', onCancel);
+    }
```

Comment:
I believe that's equivalent 

```suggestion
    signal?.removeEventListener('abort', onCancel);
```

so you probably don't need `let onCancel`

Reply:
Second parameter of removeEVentListener should not be undefined. This won't work.